### DISCUSSION TITLE
Slow Cygwin builds

### DISCUSSION THREAD

**Discussion by jkotlinski** [jkotlinski/durexforth#451] [General]
I'm building Durexforth on Cygwin + Windows 10. One annoyance is that it takes a long time simply to copy files to the .d64 image. Each c1541 call takes 1/3 of a second, and it adds up.

Any ideas how to improve this? I suspect in some parts, Cygwin is simply slow. Are there better options for Windows 10? MSYS2 seems twice as fast, is it the best bet?

```
$ time echo quit | c1541.exe
c1541 (VICE 3.6.0)
Copyright 1995-2021 The VICE Development Team.
C1541 is free software, covered by the GNU General Public License, and you are
welcome to change it and/or distribute copies of it under certain conditions.

real    0m0.350s
user    0m0.030s
sys     0m0.077s
```

**Comment by Whammo** [comment]
If fiddling with processor priority and/or affinity hasn't helped, it's a good 1541 emulation.

**Comment by lonetech** [comment]
The obvious answer would be to not call c1541 repeatedly, but pass it a script with several commands. #452 might help. Test on Linux ran in ~60ms rather than ~340ms. On the other hand, it could be a lot of the time is spent in make running each line of the recipe using a new shell. Adding [`.ONESHELL:`](https://www.gnu.org/software/make/manual/html_node/One-Shell.html) could well speed up recipes like this that run a dozen lines. 

The `time ( echo quit | c1541 )` test ran in 8ms on this machine (and amusingly, c1541 -quit takes a longer time and crashes). I've found MSYS2 to work pretty well, but they both emulate a Unix-like system on Windows. It's a painful task. I suppose some people might suggest [WSL](https://docs.microsoft.com/en-us/windows/wsl/about), but I've never tried it. Version 2 did directly target Windows' strangely poor file system performance.

**Reply by jkotlinski** [reply]
excellent tips, thank you!
